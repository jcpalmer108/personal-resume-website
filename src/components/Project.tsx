import { 
  Wrapper,
  Icon,
  Industry,
  Title,
  Description,
  Heading,
  TitleLine
} from "../styles/components/Project"

type ProjectProps = {
  icon: string,
  industry: string,
  title: string,
  description: string[]
}

export default function Project({ icon, industry, title, description }: ProjectProps) {
  return (
    <Wrapper data-testid="Project">
      <Heading>
        <Icon data-testid="Icon" src={require("../assets/images/" + icon + ".svg")} />
        <TitleLine>
          <Title data-testid="Title">{title}</Title>
          <Industry data-testid="Industry">{industry}</Industry>
        </TitleLine>
      </Heading>
      <Description data-testid="Description">
        {description.map((paragraph) => <p>{paragraph}</p>)}
      </Description>
    </Wrapper>
  )
}