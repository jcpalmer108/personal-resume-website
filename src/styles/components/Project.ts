import styled from "styled-components";
import { silver_polish, tabletBreakpoint } from "../variables";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;

  @media (min-width: ${tabletBreakpoint}px) {
    align-items: flex-start;
  }
`;

export const Icon = styled.img`
  height: 24px;
  width: 24px;
  border: 1px solid ${silver_polish};
  padding: 20px;
`;

export const Title = styled.div`
  font-family: Gilroy-Regular;
  font-size: 14px;
  font-weight: bold;
  padding-top: 15px;

  @media (min-width: ${tabletBreakpoint}px) {
    padding-top: 0px;
  }
`;
export const Industry = styled.div`
  font-family: Gilroy-Medium;
  color: ${silver_polish};
  font-size: 12px;
  padding-top: 10px;

  @media (min-width: ${tabletBreakpoint}px) {
    text-align: left;
  }
`;

export const Description = styled.div`
  text-align: left;
`;

export const Heading = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 25px;

  @media (min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
    justify-content: flex-start;
    column-gap: 20px;
  }
`;
export const TitleLine = styled.div`
  align-text: left;
`;
