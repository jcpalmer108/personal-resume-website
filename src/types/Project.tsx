export type ProjectProps = {
  icon: string;
  industry: string;
  title: string;
  description: string[];
};
