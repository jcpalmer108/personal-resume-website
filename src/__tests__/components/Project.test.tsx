import { render, screen } from "@testing-library/react";
import Project from "../../components/Project"

describe('Project', () => {
  const params = {
    icon: 'test',
    industry: 'test industry',
    title: 'test title',
    description: [ '1', '2' ]
  }

  test('renders if optional and required params are passed in', () => {
    render(<Project icon={params.icon} industry={params.industry} title={params.title} description={params.description}/>)
    expect(screen.getByTestId("Project")).toMatchSnapshot()
    expect(screen.getByTestId("Icon")).toBeTruthy()
    expect(screen.getByTestId("Title")).toHaveTextContent(params.title)
    expect(screen.getByTestId("Industry")).toHaveTextContent(params.industry)
    expect(screen.getByTestId("Description")).toHaveTextContent(params.description.join(""))

  })
})
