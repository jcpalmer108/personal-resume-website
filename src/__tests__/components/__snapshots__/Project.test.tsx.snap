// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders if optional and required params are passed in 1`] = `
<div
  class="sc-bczRLJ gXufTg"
  data-testid="Project"
>
  <div
    class="sc-jSMfEi ksSEIw"
  >
    <img
      class="sc-gsnTZi lbzuKP"
      data-testid="Icon"
      src="[object Object]"
    />
    <div
      class="sc-gKXOVf kRtupl"
    >
      <div
        class="sc-dkzDqf dukhYI"
        data-testid="Title"
      >
        test title
      </div>
      <div
        class="sc-hKMtZM ivrWzx"
        data-testid="Industry"
      >
        test industry
      </div>
    </div>
  </div>
  <div
    class="sc-eCYdqJ poQEE"
    data-testid="Description"
  >
    <p>
      1
    </p>
    <p>
      2
    </p>
  </div>
</div>
`;
